pkg/apis/core/v1beta1/defaults_shoot_test.go

```diff
@@ -631,66 +631,51 @@ var _ = Describe("Defaults", func() {
+			Expect(obj.Spec.Maintenance.TimeWindow.End).To(HaveSuffix("0000+0000"))
 		})
 	})
 
-	Describe("#SetDefaults_Worker", func() {
-		var obj *Worker
-
-		BeforeEach(func() {
-			obj = &Worker{}
-		})
-
+	Describe("Worker defaulting", func() {
 		It("should set the allowSystemComponents field", func() {
-			obj.SystemComponents = nil
-
-			SetDefaults_Worker(obj)
-
-			Expect(obj.SystemComponents.Allow).To(BeTrue())
+			SetObjectDefaults_Shoot(obj)
+			for i := range obj.Spec.Provider.Workers {
+				a := &obj.Spec.Provider.Workers[i]
+				Expect(a.SystemComponents.Allow).To(BeTrue())
+			}
 		})
 	})
 
-	Describe("#SetDefaults_ClusterAutoscaler", func() {
+	Describe("ClusterAutoscaler defaulting", func() {
 		var (
-			obj                *ClusterAutoscaler
 			expanderRandom     = ClusterAutoscalerExpanderRandom
 			expanderLeastWaste = ClusterAutoscalerExpanderLeastWaste
 		)
 
-		BeforeEach(func() {
-			obj = &ClusterAutoscaler{}
-		})
-
 		It("should correctly default the ClusterAutoscaler", func() {
-			SetDefaults_ClusterAutoscaler(obj)
+			obj.Spec.Kubernetes.ClusterAutoscaler = &ClusterAutoscaler{}
 
-			Expect(obj.ScaleDownDelayAfterAdd).To(PointTo(Equal(metav1.Duration{Duration: 1 * time.Hour})))
-			Expect(obj.ScaleDownDelayAfterDelete).To(PointTo(Equal(metav1.Duration{Duration: 0})))
-			Expect(obj.ScaleDownDelayAfterFailure).To(PointTo(Equal(metav1.Duration{Duration: 3 * time.Minute})))
-			Expect(obj.ScaleDownUnneededTime).To(PointTo(Equal(metav1.Duration{Duration: 30 * time.Minute})))
-			Expect(obj.ScanInterval).To(PointTo(Equal(metav1.Duration{Duration: 10 * time.Second})))
-			Expect(obj.MaxNodeProvisionTime).To(PointTo(Equal(metav1.Duration{Duration: 20 * time.Minute})))
-			Expect(obj.Expander).To(PointTo(Equal(expanderLeastWaste)))
-			Expect(obj.MaxGracefulTerminationSeconds).To(PointTo(Equal(int32(600))))
+			SetObjectDefaults_Shoot(obj)
+
+			Expect(obj.Spec.Kubernetes.ClusterAutoscaler.ScaleDownDelayAfterAdd).To(PointTo(Equal(metav1.Duration{Duration: 1 * time.Hour})))
+			Expect(obj.Spec.Kubernetes.ClusterAutoscaler.ScaleDownDelayAfterDelete).To(PointTo(Equal(metav1.Duration{Duration: 0})))
+			Expect(obj.Spec.Kubernetes.ClusterAutoscaler.ScaleDownDelayAfterFailure).To(PointTo(Equal(metav1.Duration{Duration: 3 * time.Minute})))
+			Expect(obj.Spec.Kubernetes.ClusterAutoscaler.ScaleDownUnneededTime).To(PointTo(Equal(metav1.Duration{Duration: 30 * time.Minute})))
+			Expect(obj.Spec.Kubernetes.ClusterAutoscaler.ScanInterval).To(PointTo(Equal(metav1.Duration{Duration: 10 * time.Second})))
+			Expect(obj.Spec.Kubernetes.ClusterAutoscaler.MaxNodeProvisionTime).To(PointTo(Equal(metav1.Duration{Duration: 20 * time.Minute})))
+			Expect(obj.Spec.Kubernetes.ClusterAutoscaler.Expander).To(PointTo(Equal(expanderLeastWaste)))
+			Expect(obj.Spec.Kubernetes.ClusterAutoscaler.MaxGracefulTerminationSeconds).To(PointTo(Equal(int32(600))))
 		})
 
 		It("should correctly override values the ClusterAutoscaler on setting them", func() {
```

Comment:
```suggestion
		It("should not overwrite already set values for ClusterAutoscaler fields", func() {
```

I think for every such struct we want to have 2 tests - one test of type `should correctly default the field` and other test of type `should not overwrite already set values`. You could also try to unify the IT names because right now they are different.